packages/cmf/src/http.js

```diff
@@ -0,0 +1,274 @@
+ *  'Accept-Language': preferredLanguage,
+ * }});
+ */
+export function setDefaultConfig(config) {
+	if (HTTP.defaultConfig) {
+		throw new Error(
+			'ERROR: setDefaultConfig should not be called twice, if you wish to change the language use setDefaultLanguage api.',
+		);
+	}
+
+	HTTP.defaultConfig = config;
+}
+
+/**
+ * To change only the Accept-Language default headers
+ * on the global http defaultConfig
+ * @param {String} language
+ */
+export function setDefaultLanguage(language) {
+	if (get(HTTP, 'defaultConfig.headers')) {
+		HTTP.defaultConfig.headers['Accept-Language'] = language;
+	} else {
+		// eslint-disable-next-line no-console
+		throw new Error('ERROR: you should call setDefaultConfig.');
+	}
+}
+
+/**
+ * getDefaultConfig - return the defaultConfig
+ *
+ * @return {object}  the defaultConfig used by cmf
+ */
+export function getDefaultConfig() {
+	return HTTP.defaultConfig;
+}
+
+/**
+ * merge the CSRFToken handling rule from the module defaultConfig
+ * into config argument
+ * @param {Object} config
+ * @returns {Function}
+ */
+export function handleCSRFToken(config) {
+	return mergeCSRFToken({
+		security: config.security,
+	})(config);
+}
+
+export class HTTPError extends Error {
+	constructor({ data, response }) {
+		super(response.statusText);
+
+		this.name = `HTTP ${response.status}`;
+		this.data = data;
+		this.response = response;
+	}
+}
+
+/**
+ * encodePayload - encore the payload if necessary
+ *
+ * @param  {object} headers request headers
+ * @param  {object} payload payload to send with the request
+ * @return {string|FormData} The encoded payload.
+ */
+export function encodePayload(headers, payload) {
+	const type = headers['Content-Type'];
+
+	if (payload instanceof FormData || typeof payload === 'string') {
+		return payload;
+	} else if (type && type.includes('json')) {
+		return JSON.stringify(payload);
+	}
+	return payload;
+}
+
+/**
+ * handleHttpResponse - handle the http body
+ *
+ * @param  {Response} response A response object
+ * @return {Promise}           A promise that resolves with the result of parsing the body
+ */
+export function handleBody(response) {
+	let methodBody = 'text';
+
+	const headers = get(response, 'headers', new Headers());
+	const contentType = headers.get('Content-Type');
+	if (contentType && contentType.includes('application/json')) {
+		methodBody = 'json';
+	}
+
+	if (contentType && contentType.includes('application/zip')) {
+		methodBody = 'blob';
+	}
+
+	return response[methodBody]().then(data => ({ data, response }));
+}
+
+/**
+ * handleHttpResponse - handle the http error
+ *
+ * @param  {Response} response A response object
+ * @return {Promise}           A promise that reject with the result of parsing the body
+ */
+export function handleError(response) {
+	// in case of network issue
+	if (response instanceof Error) {
+		return new HTTPError({ response, data: response });
+	}
+	return handleBody(response).then(body => new HTTPError(body));
```

Comment:
can we stop swallowing errors?

Reply:
Sure we can but it will be a breaking change with the current API
At least, we can think about it.
It may be an idea to expose this tools in a separate package as it feels a bit weird to me too to import http from react-cmf, if we do so, we can keep the swallowing with the current implementation in cmf and let the error be dispatch in the new package.

WDYT @jsomsanith-tlnd  ?

Reply:
Let's keep the scope of this PR which is extracting the code to be usable. Agree with you about let the error be dispatched in the extracted code.
For the existing api, feel free to create a new PR and we'll schedule the breaking change properly